awx/main/models/inventory.py

```diff
@@ -1404,5 +1405,10 @@ def get_absolute_url(self, request=None):
         return reverse('api:inventory_script_detail', kwargs={'pk': self.pk}, request=request)
 
 
-for cls in PluginFileInjector.__subclasses__():
-    InventorySourceOptions.injectors[cls.__name__] = cls
+awx_entry_points = {ep.name: ep for ep in entry_points(group='awx_plugins.inventory')}
+supported_entry_points = {ep.name: ep for ep in entry_points(group='awx_plugins.inventory.supported')}
+entry_points = awx_entry_points if detect_server_product_name() == 'AWX' else {**awx_entry_points, **supported_entry_points}
```

Comment:
Perhaps, it's better to skip the lookup altogether when not needed?
```suggestion
awx_entry_points = {ep.name: ep for ep in entry_points(group='awx_plugins.inventory')}
supported_entry_points = dict() if detect_server_product_name() == 'AWX' else {ep.name: ep for ep in entry_points(group='awx_plugins.inventory.supported')}
entry_points = {**awx_entry_points, **supported_entry_points}
```